packages/hardhat-core/src/internal/hardhat-network/provider/node.ts

```diff
@@ -358,6 +362,25 @@ export class HardhatNode extends EventEmitter {
     return new BN((await this.getLatestBlock()).header.number);
   }
 
+  public async getPendingBlock(): Promise<Block> {
+    const result = await this._runInBlockContext("pending", () =>
+      this._blockchain.getLatestBlock()
+    );
+
+    return result;
+  }
+
+  public async getPendingBlockAndTotalDifficulty(): Promise<[Block, BN]> {
+    return this._runInBlockContext("pending", async () => {
+      const block = await this._blockchain.getLatestBlock();
+      const totalDifficulty = await this._blockchain.getTotalDifficulty(
+        block.hash()
+      );
+
+      return [block, totalDifficulty];
+    });
+  }
```

Comment:
Instead of adding `getPendingBlock` method I would make `getBlockByNumber` accept `blockNumberOrPending: BN | "pending"` param like the other methods.

Also, (look at https://github.com/EthWorks/buidler/pull/8#discussion_r528876684 first) when you extract a `_runInPendingBlockContext` method you'll be able to use it instead of `_runInBlockContext`.